Lib/test/test_tempfile.py

```diff
@@ -982,6 +982,35 @@ def test_basic(self):
         f = self.do_create(max_size=100, pre="a", suf=".txt")
         self.assertFalse(f._rolled)
 
+    def test_iobase_abstract(self):
+        # SpooledTemporaryFile should implement the IOBase abstract
+        iobase_abstract = (
+            'close',
+            'closed',
+            'fileno',
+            'flush',
+            'isatty',
+            'read',
+            'readable',
+            'readinto',
+            'readline',
+            'readlines',
+            'seek',
+            'seekable',
+            'tell',
+            'truncate',
+            'write',
+            'writable',
+            'writelines',
+            '__del__',
+        )
+        f = self.do_create()
+        for attribute in iobase_abstract:
+            self.assertTrue(
+                hasattr(f, attribute),
+                '{} attribute missing'.format(attribute)
+            )
```

Comment:
Minor: this test could be make simpler/faster with sets (assert that set(dir(f)) is a superset of set(dir(IOBase)).

Could you also add a test with isinstance to validate that the class implements IOBase?